torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -612,19 +617,42 @@ Value* emitBuiltinCall(
     Symbol name,
     at::ArrayRef<NamedValue> args,
     at::ArrayRef<NamedValue> kwargs,
-    const c10::optional<NamedValue>& self) {
+    const c10::optional<NamedValue>& self,
+    const c10::optional<int64_t> version) {
   const auto& variants = getAllOperatorsFor(name);
   const auto& builtin_functions = getAllBuiltinFunctionsFor(name);
 
+  // first let's set the graph's version
+  if (version.has_value()) {
+    if (graph.get_op_version().has_value()) {
+      TORCH_INTERNAL_ASSERT(version.value() == graph.get_op_version().value());
+    } else {
+      graph.set_op_version(version.value());
+    }
+  }
+
   std::stringstream failure_messages;
-  std::vector<const FunctionSchema*> schemas;
+  std::vector<FunctionSchema> schemas;
   schemas.reserve(variants.size());
   for (const std::shared_ptr<Operator>& op : variants) {
-    schemas.push_back(&op->schema());
+    auto op_name =
+        op->schema().operator_name().name + "." + op->schema().overload_name();
+    if (version.has_value()) {
+      auto version_entry = operator_version_map.find(op_name);
+      if (version_entry != operator_version_map.end()) {
+        auto old_schema_entry =
+            findUpgrader(version_entry->second, version.value());
+        auto old_schema = parseSchema(old_schema_entry.old_schema);
```

Comment:
I was wondering why we change `matchSchemas` to own `FunctionSchema` when reading the code...  seems like you want to construct new FunctionSchema from old entries. Not sure if you're working on refactoring this, but I think we do need to cache these schemas in a singleton in version_maps, and you don't need to change `const FunctionSchema*` to `FunctionSchema` and saves lots of copies.